File: packages/node/webrtc-client/src/shared/tracing/tracer.ts
import { WebTracerProvider } from "@opentelemetry/sdk-trace-web";
import { BatchSpanProcessor } from "@opentelemetry/sdk-trace-web";
import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-http";
import { Resource } from "@opentelemetry/resources";
import { SEMRESATTRS_SERVICE_NAME, SEMRESATTRS_SERVICE_VERSION } from "@opentelemetry/semantic-conventions";
import { trace } from "@opentelemetry/api";

export function initTracer(): void {
  const resource = new Resource({
    [SEMRESATTRS_SERVICE_NAME]: "node.webrtc-client",
    [SEMRESATTRS_SERVICE_VERSION]: "1.0.0",
  });

  const provider = new WebTracerProvider({ resource });

  const otlpEndpoint = import.meta.env["VITE_OTEL_ENDPOINT"] as string | undefined;
  if (otlpEndpoint) {
    provider.addSpanProcessor(
      new BatchSpanProcessor(new OTLPTraceExporter({ url: otlpEndpoint }))
    );
  }

  provider.register();
}

export function getTracer() {
  return trace.getTracer("webrtc-client", "1.0.0");
}

File: packages/python/webrtc-signaling/src/shared/tracing/tracer.py
import os
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
from opentelemetry.sdk.resources import Resource, SERVICE_NAME, SERVICE_VERSION
from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter


def init_tracer() -> trace.Tracer:
    resource = Resource.create(
        {
            SERVICE_NAME: "python.webrtc-signaling",
            SERVICE_VERSION: "1.0.0",
            "deployment.env": os.getenv("DEPLOYMENT_ENV", "dev"),
        }
    )
    provider = TracerProvider(resource=resource)

    if os.getenv("SKIP_CONSOLE_EXPORTER") != "true":
        provider.add_span_processor(BatchSpanProcessor(ConsoleSpanExporter()))

    otlp_endpoint = os.getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
    if otlp_endpoint:
        provider.add_span_processor(BatchSpanProcessor(OTLPSpanExporter(endpoint=otlp_endpoint)))

    trace.set_tracer_provider(provider)
    return trace.get_tracer("webrtc-signaling", "1.0.0")
